### DISCUSSION TITLE
How to force single quotes by passing an argument?

### DISCUSSION THREAD

**Discussion by suhaybu** [astral-sh/ruff#9743] [Q&A] [answered]
I'm a newbie who's been a black formatter enjoyer (using the VS Code extension) that discovered Ruff just today. However, I prefer single quotes over double quotes.

**Is there a way I can force single quotes over double quotes without creating a .toml file?**

Going through the documentation and information I could find about switching from double quotes to single quotes, the right way to do it according to the [docs](https://docs.astral.sh/ruff/formatter/#configuration) is as follows:

#### for pyproject.toml
```toml
[tool.ruff.format]
quote-style = "single"
```

#### for ruff.toml
```toml
[format]
quote-style = "single"
```

When using the black formatter, I was able to achieve the same without making a toml file by passing `--single-quotes` as an argument.

My VS Code's settings.json had this line.
```json
"python.formatting.blackArgs": ["--single-quotes"],
```

Is there any way I can force single quote without creating a toml file when using Ruff? In my mind I had these two possibilities for args:

Maybe like this
```json
"ruff.format.args": ["--single-quotes"]
```

Or like this
```json
"ruff.format.args": ["--quote-style", "single"]
```

> [!NOTE]
> I didn't put this in a feature request because I'm not entirely sure if I've overlooked something or if such a feature doesn't actually exist.

**Comment by charliermarsh** [comment, marked as answer]
Thanks for the thorough issue! This actually doesn't exist yet, but it's about to exist: https://github.com/astral-sh/ruff/pull/9599. We're adding a mechanism to provide any setting / configuration option via the CLI.

**Reply by suhaybu** [reply]
That's awesome! It'll cover all use cases in that case.

Good luck and thank you for the speedy response. 👍

**Reply by AlexWaygood** [reply]
Yes! Specifically, once #9599 is merged, you'll be able to specify any configuration option from the CLI using "inline TOML". So to specify single-quotes for the formatter from the CLI, you'd use this command:

```sh
ruff format path/to/code --config "format.quote-style = 'single'"
```
